src/main/java/com/github/dockerjava/netty/InvocationBuilder.java

```diff
@@ -391,15 +391,18 @@ private void setDefaultHeaders(HttpRequest request) {
 
         Channel channel = getChannel();
 
-        JsonResponseCallbackHandler<T> jsonResponseHandler = new JsonResponseCallbackHandler<T>(typeReference,
-                resultCallback);
-
         HttpResponseHandler responseHandler = new HttpResponseHandler(requestProvider, resultCallback);
 
         channel.pipeline().addLast(new ChunkedWriteHandler());
         channel.pipeline().addLast(responseHandler);
-        channel.pipeline().addLast(new JsonObjectDecoder());
-        channel.pipeline().addLast(jsonResponseHandler);
+
+        if (!Void.class.equals(typeReference.getType())) {
```

Comment:
I would rather refractor and extract a specific method without JSON mapping. I'll do it in the nearest days.